src/main/java/Duke.java

```diff
@@ -1,10 +1,56 @@
 public class Duke {
+
+    private UI ui;
+
+   public String logo = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+
+    /**
+     * Public constructor of Duke.
+     */
+    public Duke() {
+        this.ui = new UI();
+    }
+
+    /**
+     * Invoke run for Duke chatbot programme.
+     *
+     * @param args argument.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke().run();
+    }
+
+    /**
+     * Run Duke programme depending on the different inputs/commands
+     * given by user.
+     */
+    public void run() {
+        System.out.println(logo + "\n" + ui.greet());
+        boolean stillRunning = true;
+        while (stillRunning) {
+            try {
+                String input = ui.getInput();
+                if (input.equals("bye")) {
+                    System.out.println(ui.bye());
+                    stillRunning = false;
+                } else if (input.equals("list")) {
+                    System.out.println(ui.list());
+                } else if (input.startsWith("mark") || input.startsWith("unmark")) {
+                    System.out.println(ui.editTask(input));
+                } else if (input.startsWith("todo ") || input.startsWith("event ") || input.startsWith("deadline ")){
+                    System.out.println(ui.addTask(input));
+                } else if (input.startsWith("delete")) {
+                    System.out.println(ui.deleteTask(input));
+                }else {
```

Comment:
```suggestion
                } else if (input.startsWith("todo ") || input.startsWith("event ") || input.startsWith("deadline ")) {
                    System.out.println(ui.addTask(input));
                } else if (input.startsWith("delete")) {
                    System.out.println(ui.deleteTask(input));
                } else {
```
>  White space within a statement

You are missing some whitespaces here